api/rosetta/data_integration_test.go

```diff
@@ -0,0 +1,269 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// +build integration
+
+package rosetta_test
+
+import (
+	"encoding/hex"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/klauspost/compress/zstd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/api/rosetta"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/configuration"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/optakt/flow-dps/rosetta/invoker"
+	"github.com/optakt/flow-dps/rosetta/retriever"
+	"github.com/optakt/flow-dps/rosetta/scripts"
+	"github.com/optakt/flow-dps/rosetta/validator"
+	"github.com/optakt/flow-dps/service/dictionaries"
+	"github.com/optakt/flow-dps/service/index"
+	"github.com/optakt/flow-dps/testing/snapshots"
+)
+
+const (
+	invalidBlockchain = "invalid-blockchain"
+	invalidNetwork    = "invalid-network"
+	invalidToken      = "invalid-token"
+
+	invalidBlockHash = "af528bb047d6cd1400a326bb127d689607a096f5ccd81d8903dfebbac26afb2z" // invalid hex value
+
+	validBlockHashLen = 64
+)
+
+func setupDB(t *testing.T) *badger.DB {
+	t.Helper()
+
+	opts := badger.DefaultOptions("").
+		WithInMemory(true).
+		WithReadOnly(true).
```

Comment:
Hm, yeah, not sure how much sense it makes with an in-memory database.. I wanted to make the intent clear, that tests do not alter anything in the DB

Reply:
I don't think that's necessary in test code, since every database is throw-away anyway.

Reply:
It does ensure that code that isn't supposed to write on the DB can't do so, though. If we ever accidentally try to write on the db, this will catch it by panicking

Reply:
The problem is that, if Badger was consistently implemented, loading the snapshot into a read-only DB would fail.